manifests/init.pp

```diff
@@ -49,47 +58,59 @@
   }
 
   podman::secret { $server_cert_secret_name:
-    path => $certs::iop_advisor_engine::server_cert,
+    ensure => $ensure,
+    path   => $certs::iop_advisor_engine::server_cert,
   }
 
   podman::secret { $server_key_secret_name:
-    path => $certs::iop_advisor_engine::server_key,
+    ensure => $ensure,
+    path   => $certs::iop_advisor_engine::server_key,
   }
 
   podman::secret { $server_ca_cert_secret_name:
-    path => $certs::iop_advisor_engine::server_ca_cert,
+    ensure => $ensure,
+    path   => $certs::iop_advisor_engine::server_ca_cert,
   }
 
   podman::secret { $client_cert_secret_name:
-    path => $certs::iop_advisor_engine::client_cert,
+    ensure => $ensure,
+    path   => $certs::iop_advisor_engine::client_cert,
   }
 
   podman::secret { $client_key_secret_name:
-    path => $certs::iop_advisor_engine::client_key,
+    ensure => $ensure,
+    path   => $certs::iop_advisor_engine::client_key,
   }
 
   podman::secret { $client_ca_cert_secret_name:
-    path => $certs::iop_advisor_engine::client_ca_cert,
+    ensure => $ensure,
+    path   => $certs::iop_advisor_engine::client_ca_cert,
+  }
+
+  $service_ensure = $ensure ? {
+    'present' => true,
+    'absent'  => false,
   }
 
   file { $log_dir:
-    ensure => directory,
+    ensure => $directory_ensure,
     mode   => '0755',
     owner  => 'root',
     group  => 'root',
   }
 
   ['uploads', 'failed', 'logs'].each |String $file| {
     file { "${log_dir}/${file}":
-      ensure => directory,
+      ensure => $directory_ensure,
+      force  => true,
       mode   => '0755',
       owner  => 'root',
       group  => 'root',
     }
   }
```

Comment:
I think ordering those absents will fail (I recall it will try to order log_dir before log_dir/something, and boom.

Reply:
interestingly, it does not, so whatever